lib/spack/spack/version.py

```diff
@@ -56,6 +56,9 @@
 # Valid version characters
 VALID_VERSION = r'[A-Za-z0-9_.-]'
 
+# Infinity-like versions. The order in the list implies the comparision rules
+infinity_versions = ['develop']
```

Comment:
@adamjstewart 

> Not to open up a new can of worms, but what about master?

this line can be changed into 
```
infinity_versions = ['develop', 'master', 'head', 'trunk']
```
if there is a consensus to do that. Everything will follow automatically.